### ISSUE TITLE
[NEW] Sending attachments should be done in background rather than showing loading

### ISSUE THREAD

**Issue by divyanshub024** [RocketChat/Rocket.Chat.Android#943]

**Comment by philipbrito** [comment]
What do you mean with "done in background"? Is it about the background thread? Could you elaborate your issue please?

**Comment by divyanshub024** [comment]
Yes, Its about the background thread. So when you upload a file it will show loading and after the loading, it will send the attachment. What I suggest is once you select a file to send it should update the UI and in the background thread you should upload it.

**Comment by rafaelks** [comment]
@divyanshub024 I'm not sure I followed the description of this issue... the upload is being done in a background thread. Can you explain better please?

**Comment by divyanshub024** [comment]
@rafaelks Yes, upload is being done in the background but its showing loading until its uploaded. It may be fine in good network connection but in poor network condition, you have to see loading for quite some time. So what I suggest is to show loading only on that image which is being uploaded rather than showing loading at the centre of screen.

**Comment by leonardoaramaki** [comment]
@divyanshub024 I agree. I'm already working on it with a better upload feedback.

**Comment by lorek123** [comment]
any updates? Still if you for example rotate the screen upload aborts.

**Comment by bizzbyster** [comment]
@leonardoaramaki any progress on this? I completely agree with @divyanshub024 . Beyond this we need guaranteed attachment delivery just like we have for messages. So when an attachment is uploaded by a user it is first stored locally. And then in the background we upload. A successful upload will be indicated graphically using changing checkmarks.

@filipedelimabrito @Shailesh351 @kb0304

**Comment by philipbrito** [comment]
@bizzbyster No one is working on it currently but everyone is free to send us a PR.
